Assembly-CSharp/ModLoader.cs

```diff
@@ -206,7 +207,12 @@ Assembly Resolve(object sender, ResolveEventArgs args)
             GetPreloads(orderedMods, scenes, toPreload);
             if (toPreload.Count > 0)
             {
-                yield return PreloadScenes(coroutineHolder, toPreload, preloadedObjects);
+                Preloader pld = coroutineHolder.GetComponent<Preloader>();
+                if (pld == null)
+                {
+                    pld = coroutineHolder.AddComponent<Preloader>();
+                }
```

Comment:
Do you think it's worth making this its own extension method (something like `GetOrAddComponent<T>`)? Given that the MAPI uses it and it's like useful all the time in mods

Reply:
(And at least three library mods currently on Scarab have their own version)

Reply:
i also just added it because i remembered that it might be added twice since the debugview thing that is used for modloading loads twice.